linkerd/app/outbound/src/endpoint.rs

```diff
@@ -68,64 +67,52 @@ impl<E> Outbound<E> {
     }
 
     /// Pushes a layer that checks if a discovered service profile contains an
-    /// endpoint override, and forwards directly to that endpoint (bypassing the
-    /// current stack) if one exists.
-    pub fn push_endpoint_override<T, O, R>(
+    /// endpoint, and forwards directly to that endpoint (bypassing the current
+    /// stack) if one exists.
+    pub fn push_profile_endpoint<T, E, ESvc, NSvc, Req>(
         self,
-        ep_override: O,
+        endpoint: E,
     ) -> Outbound<
         impl svc::NewService<
                 (Option<profiles::Receiver>, T),
-                Service = OrOverride<E::Service, O::Service>,
+                Service = svc::stack::ResultService<svc::Either<NSvc, ESvc>>,
             > + Clone,
     >
     where
-        E: svc::NewService<(Option<profiles::Receiver>, T)> + Clone,
-        E::Service: svc::Service<R, Error = Error>,
-        <E::Service as svc::Service<R>>::Future: Send,
-        O: svc::NewService<ProfileOverride> + Clone,
-        O::Service:
-            svc::Service<R, Response = <E::Service as svc::Service<R>>::Response, Error = Error>,
-        <O::Service as svc::Service<R>>::Future: Send,
-        T: std::fmt::Debug,
+        N: svc::NewService<(Option<profiles::Receiver>, T), Service = NSvc> + Clone,
+        NSvc: svc::Service<Req, Error = Error>,
+        NSvc::Future: Send,
+        E: svc::NewService<ProfileEndpoint, Service = ESvc> + Clone,
+        ESvc: svc::Service<Req, Response = NSvc::Response, Error = Error>,
+        ESvc::Future: Send,
```

Comment:
FWIW, I didn't use separate type parameters for the service types because I noticed that sometimes `push_xxx` methods' target types can't be inferred, and reducing the number of type params means fewer `_, _, _,`s have to be written out at the callsite. This is absolutely not a big deal, so I'm happy to use this style if we prefer it — the `where` clause certainly looks nicer.